presto-cache/src/main/java/com/facebook/presto/cache/CacheFactory.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.cache;
+
+import com.facebook.presto.hive.filesystem.ExtendedFileSystem;
+import org.apache.hadoop.conf.Configuration;
+
+import java.net.URI;
+
+public interface CacheFactory
```

Comment:
These changes are borrowed from my previous PR. I earlier thought it might be used at multiple places, but I guess its not the case. 
Lets remove this interface, as it is only implemented once.

Reply:
Removed.

Reply:
Actually I take this back, we would beed this factory to have its own implementation for facebook (prism) side. So let's keep the interface as it is(I don't think the interface is removed in the latest changes anyway). @highker  Let me know if you see any concerns.